src/components/GoalsComponents/MyGoal/MyGoal.tsx

```diff
@@ -20,6 +21,9 @@ interface MyGoalProps {
 const MyGoal: React.FC<MyGoalProps> = ({ goal, dragAttributes, dragListeners }) => {
   const [expandGoalId, setExpandGoalId] = useState("root");
   const [isAnimating, setIsAnimating] = useState(true);
+  const { title } = goal;
+  const titleContainsVideoLink = title.includes("youtube") || title.includes("peertube") || title.includes("youtu");
```

Comment:
this logic is for dropdown component so we can move it there. since we already have the goal requirement in props. you can do this same processing there.